pkg/cloudprovider/vsphereparavirtual/apis/nsxnetworking/v1alpha1/routetable_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RouteTable describe a routing table resource.
+type RouteTable struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   RouteTableSpec   `json:"spec"`
+	Status RouteTableStatus `json:"status,omitempty"`
+}
+
+// RouteTableSpec defines the desired state of RouteTable.
+type RouteTableSpec struct {
+	// Routes is the list of desired routes.
+	Routes []Route `json:"routes"`
+}
+
+// Route defines a route entry for traffic.
+type Route struct {
+	// Name defines the name of this route.
+	Name string `json:"name"`
+	// Destination is the CIDR block used for the destination match.
+	Destination string `json:"destination"`
+	// Target determines where to target the traffic.
+	Target string `json:"target"`
+}
+
+// RouteTableStatus defines the observed state of RouteTable.
+type RouteTableStatus struct {
+	// Routes is the list of realized route.
+	Routes []Route `json:"routes"`
```

Comment:
@nicolehanjing which do you think is better about the way to put realized routes in status? either to put the complete `Route` list in status, or just put names of Route?

Reply:
if I am understanding it correctly, I believe we only add route CR items that are ready/healthy
I am a little confused about the mapping relationship here:
1. each item in []Route should have the corresponding Condition in []RouteTableCondition, right? then should we name it []RouteCondition instead?
2. I see we remove the "failure" condition, previously I thought the mapping looks like this:
```
route1 | ready
route2 | failure
route3 | ready
```
if we only return "ready" routes, how should we map the route and its condition?
```
route1 | ready
route2 | ready (means route 3 is ready)
route3 | 
```

Reply:
`RouteCondition` is a list, but not for each `Route`, it is the overall status for `RouteSet` CR.